src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,234 @@
+            }
+
+            return builder.ToImmutable();
+        }
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var position = context.Position;
+            var syntaxTree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var token = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
+            token = token.GetPreviousTokenIfTouchingWord(position);
+            if (!token.IsKind(SyntaxKind.DotToken))
+            {
+                return;
+            }
+
+            var expression = GetParentExpressionOfInvocation(token);
+            if (expression is null)
+            {
+                return;
+            }
+
+            var semanticModel = await document.ReuseExistingSpeculativeModelAsync(expression.SpanStart, cancellationToken).ConfigureAwait(false);
+            var container = semanticModel.GetTypeInfo(expression, cancellationToken).Type;
+            if (container is null)
+            {
+                return;
+            }
+
+            var allMembers = container.GetMembers();
+            var allExplicitConversions = from m in allMembers.OfType<IMethodSymbol>()
+                                         where
+                                             m.IsConversion() && // MethodKind.Conversion
+                                             m.Name == WellKnownMemberNames.ExplicitConversionName && // op_Explicit
+                                             container.Equals(m.Parameters[0].Type) // Convert from container type to other type
+                                         let typeName = m.ReturnType.ToMinimalDisplayString(semanticModel, position)
+                                         select SymbolCompletionItem.CreateWithSymbolId(
+                                             displayText: $"({typeName})", // The type to convert to
+                                             filterText: typeName,
+                                             sortText: $"{SortingPrefix}{typeName}",
+                                             symbols: ImmutableList.Create(m),
+                                             rules: CompletionItemRules.Default,
+                                             contextPosition: position,
+                                             properties: CreateCompletionHandlerProperty(CompletionHandlerConversion, (MinimalTypeNamePropertyName, typeName)));
+            context.AddItems(allExplicitConversions);
+
+            var indexers = allMembers.OfType<IPropertySymbol>().Where(p => p.IsIndexer).ToImmutableList();
+            if (!indexers.IsEmpty)
+            {
+                var indexerCompletion = SymbolCompletionItem.CreateWithSymbolId(
+                               displayText: $"this[]",
+                               filterText: "[",
```

Comment:
interesting. not sure if this is a good idea.  typing `this` to filter down to the exposed indexers seems reasonable.

Reply:
Changed to "this" in f1a17bbd. (The brace didn't work while I was testing live).